pkg/sdk/errors.go

```diff
@@ -48,3 +47,230 @@ func decodeDriverError(err error) error {
 
 	return err
 }
+
+const ghIssueBodyTemplate = `
+**Provider Version**
+
+The provider version you are using.
+
+**Terraform Version**
+
+The version of Terraform you were using when the bug was encountered.
+
+**Describe the bug**
+
+A clear and concise description of what the bug is.
+
+**Expected behavior**
+
+A clear and concise description of what you expected to happen.
+
+**Code samples and commands**
+
+Please add code examples and commands that were run to cause the problem.
+
+**Additional context**
+
+Add any other context about the problem here.
+
+<!-- 
+Please provide additional error messages if we missed any (see the errors below and compare it with your console output)
+-->
+
+Errors:
+%s
+`
+
+// NewTopLevelError wraps an error with "final" error message of sdk.
+// It should be placed in the highest place of call stack to catch as much error context as possible.
+// TODO Is it possible to call this function somewhere high in the callstack, so it always be top level ? If not maybe this would be overkill / too hard to use
+// TODO It's possible to wrap errors multiple times, this function will try to keep only the one with most error context.
+func NewTopLevelError(err error) error {
+	// TODO if called multiple times, unwrap lower level errors and wrap err to have more context
+
+	return fmt.Errorf(`
+Snowflake Terraform Provider error!
+If you think you've encountered a bug, please report it with the link below.
+If any of the error information is missing in the issue body, please fill it up.
+Any additional information (or context what you were trying to achieve) would be helpful
+to provide the solution or fix as soon as possible. Thanks :)
+
+https://github.com//Snowflake-Labs/terraform-provider-snowflake/issues/new?labels=bug&title=%s&body=%s
+
+Errors:
+%s`,
+		url.QueryEscape("New issue"),
+		url.QueryEscape(fmt.Sprintf(ghIssueBodyTemplate, err.Error())),
+		err.Error(),
+	)
+}
+
+type SDKError struct { //nolint:all
```

Comment:
i don't like SDKError as a type name. because the package is SDK. so it would be sdk.SDKError, which is awkward. Would be better to rename this to SnowflakeError or possibly SnowflakeSDKError

Reply:
Changed to sdk.Error